### PR TITLE
Vehicle projectile origin adjustment

### PR DESCRIPTION
**Pull request by Doubleumc**
# About the pull request

Vehicle projectiles now start on the vehicle and travel outward, without hitting the vehicle. As a side effect, `Adjacent()` checks now accept a list of objects to ignore.

Origins are adjusted to the point visually closest to the hardpoint's sprite. For most hardpoints this is the center of the vehicle, with the only exceptions being the APC weapons (all along different edges) and the secondary flamer (flame origin is walked until not on the vehicle).

Since projectiles now must travel farther to reach the front of the vehicle, some shorter ranged ammos are adjusted to compensate.

The hardpoint check `clear_los()` is removed since now all origins are inside the vehicle's bounds.

Removed redundant call to `display_ammo()` when firing secondary flamer.

# Explain why it's good for the game

Projectiles originating from the tile in front of a vehicle make it extremely hard or even impossible to engage targets adjacent to it. With this change the projectile, and resulting cone of fire, starts where you would expect it to.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
qol: vehicle weapon projectiles start from the weapon
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -357,8 +357,12 @@
 	if((speed * world.tick_lag) >= get_dist(current_turf, target_turf))
 		SEND_SIGNAL(src, COMSIG_BULLET_TERMINAL)
 
+
+	var/obj/item/hardpoint/hardpoint = shot_from
+	var/list/ignore_list = istype(hardpoint) ? list(hardpoint.owner) : null //if fired from a vehicle, exclude the vehicle's body from the adjacency check
```
Comment (Zonespace27) [inline code/modules/projectiles/projectile.dm:362]: shouldn't `null` be an empty list instead?

File: code/modules/vehicles/hardpoints/secondary/flamer.dm [hunk 2]
```diff
@@ -29,14 +27,28 @@
...
+		return NONE
+
+	return ..()
+
 /obj/item/hardpoint/secondary/small_flamer/handle_fire(atom/target, mob/living/user, params)
-	var/turf/origin_turf = get_origin_turf()
+	//step forward along path so flame starts outside hull
+	var/list/turfs = get_line(get_origin_turf(), get_turf(target))
+	var/turf/origin_turf
+	for(var/turf/turf in turfs)
```
Comment (Zonespace27) [inline code/modules/vehicles/hardpoints/secondary/flamer.dm:41]: `as anything`?

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 1 above)
Comment (Doubleumc) [reply code/modules/projectiles/projectile.dm:362]: It doesn't have to be. Everything works just fine with `ignore_list = null`. I figured it was preferable to only initialize if needed. I'll rewrite it to make it clearer.

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**